adapters/avail/src/service.rs

```diff
@@ -252,3 +300,20 @@ impl DaService for DaProvider {
         Ok(())
     }
 }
+
+/// Used to keep channel open even if there's no other consumers
+fn run_finalized_header_consumer(mut receiver: Receiver<AvailHeader>) {
```

Comment:
TLDR: It can be removed, as discussed with Blazej yesterday.

To clarify. In the implementation, subcription to finalized headers starts at the initialization of Client, so there should be background task that will do 2 things:

1. Keep channel open for new subsribers
2. Keep channel not filled to the end, until actual subcribers appear.


But if `subscribe_finalized_header` takes `&mut self`, then it can have an option and only subcribe when first `subscribe_finalized_header` is called.